### ISSUE TITLE
Upgrade Jenkins X repositories to go 1.19

### ISSUE THREAD

**Issue by ankitm123** [jenkins-x/jx#8433]
### When we were upgrading the go version to 1.18, we noticed that we are limited by the version of go in the goreleaser image. We are now building a custom image where we can set the version of go and install other packages we need. Once that is done, we can start this update work.

Golang supports only 2 major releases: https://endoflife.date/go
The latest version of golang is 1.19.2

Follow this guide for information on how to upgrade: https://jenkins-x.io/community/maintainer_guide/go-upgrades/

Before upgrading , we need to upgrade the version of golang in the catalog repository: 
example (there are other files as well): https://github.com/jenkins-x/jx3-pipeline-catalog/blob/0f0f956d38ffe383ba38d93c1cfe7160a3bec93a/tasks/go/pullrequest.yaml#L37

This is the tentative list of repositories (list will be updated if we find any missing repositories) that we should upgrade:

- [x] [jx-kubeclient](https://github.com/jenkins-x/jx-kube-client) - no other jx dependencies
- [x] [go-scm](https://github.com/jenkins-x/go-scm) - no other jx dependencies
- [x] [logrus-stackdriver-formatter](https://github.com/jenkins-x/logrus-stackdriver-formatter) - no jx dependencies
- [ ] [lighthouse](https://github.com/jenkins-x/lighthouse)
  - go-scm
- [ ] [jx-logging](https://github.com/jenkins-x/jx-logging):
  - logrus-stackdriver-formatter
- [x] [ secretfacade](https://github.com/jenkins-x-plugins/secretfacade)
  - jx-logging
- [x] [jx-api](https://github.com/jenkins-x/jx-api)
  - jx-logging
- [ ] [jx-helpers](https://github.com/jenkins-x/jx-helpers)
  - github.com/jenkins-x/go-scm
  - github.com/jenkins-x/jx-api
  - github.com/jenkins-x/jx-kube-client
  - github.com/jenkins-x/jx-logging
- [ ] [jx](https://github.com/jenkins-x/jx/)
  - jx-api
  - jx-helpers
  - jx-kube-client
  - jx-logging
- [ ] [jx-git-operator](https://github.com/jenkins-x/jx-git-operator)
- [x] [bdd-jx3](https://github.com/jenkins-x/bdd-jx3)
- [x] [jx-gitops](https://github.com/jenkins-x-plugins/jx-gitops)
- [ ] [jx-changelog](https://github.com/jenkins-x-plugins/jx-changelog)
- [ ] [jx-secret](https://github.com/jenkins-x-plugins/jx-secret)
- [ ] [jx-admin](https://github.com/jenkins-x-plugins/jx-admin)
- [x] [jx-preview](https://github.com/jenkins-x-plugins/jx-preview)
- [ ] [jx-promote](https://github.com/jenkins-x-plugins/jx-promote)
- [ ] [jx-pipeline](https://github.com/jenkins-x-plugins/jx-pipeline)
- [ ] [jx-updatebot](https://github.com/jenkins-x-plugins/jx-updatebot)
- [ ] [jx-scm](https://github.com/jenkins-x-plugins/jx-scm)
- [ ] [jx-registry](https://github.com/jenkins-x-plugins/jx-registry)
- [ ] [ jx-slack](https://github.com/jenkins-x-plugins/jx-slack)
- [ ] [jx-release-version](https://github.com/jenkins-x-plugins/jx-release-version)
- [ ] [jx-verify](https://github.com/jenkins-x-plugins/jx-verify)
- [ ] [jx-application](https://github.com/jenkins-x-plugins/jx-application)
- [ ] [jx-build-controller](https://github.com/jenkins-x-plugins/jx-build-controller)
- [ ] [jx-test](https://github.com/jenkins-x-plugins/jx-test)

In the list, there are some packages which need multiple packages updated before we update them, for example: before upgrading jx-api we should upgrade jx-logging.

**NOTE: it's fine if you dont follow the order, the order is only there to limit the number of PRs we open as part of an upgrade effort**

We will this issue to keep track of which repositories have been upgraded.
If you are interested in contributing, please comment
```
I want to upgrade <repo-name>
```

When opening a PR in these repos to upgrade the go version, remember to include
```
related to #<issue-number>
```
Dont use `fixes` as that will close this issue once ur PR is merged.
Once your PR is merged, comment in this issue:
```
<reponame> has been upgraded to 1.19.2
```

**Comment by amarps** [comment]
I want to upgrade [jx-kubeclient](https://github.com/jenkins-x/jx-kube-client)

**Comment by KiranSatyaRaj** [comment]
I want to upgrade [go-scm](https://github.com/jenkins-x/go-scm)

**Comment by ankitm123** [comment]
I want to upgrade jx-preview

**Comment by KiranSatyaRaj** [comment]
[go-scm](https://github.com/jenkins-x/go-scm) Go version has been upgraded to 1,19.3 and it's golangci-lint to 1.50.1

**Comment by KiranSatyaRaj** [comment]
I want to upgrade [logrus-stackdriver-formatter](https://github.com/jenkins-x/logrus-stackdriver-formatter)

**Comment by PratikforCoding** [comment]
i want to upgrade jx-logging

**Comment by PratikforCoding** [comment]
I want to upgrade secretfacade

**Comment by PratikforCoding** [comment]
I want to upgrade jx-api

**Comment by PratikforCoding** [comment]
I want to upgrade jx

**Comment by PratikforCoding** [comment]
I want to upgrade jx-git-operator

**Comment by PratikforCoding** [comment]
I want to upgrade bdd-jx3

**Comment by PratikforCoding** [comment]
I want to upgrade jx-gitops

**Comment by PratikforCoding** [comment]
I want to upgrade jx-secret

**Comment by PratikforCoding** [comment]
I want to upgrade jx-admin

**Comment by PratikforCoding** [comment]
jx-logging has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
secretfacade has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
jx-api has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
jx-helpers has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
bdd-jx3 has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
jx-gitops has been upgraded to 1.19.3

**Comment by PratikforCoding** [comment]
jx-changelog has been upgraded to 1.19.3

**Comment by SrutiMi** [comment]
I want to upgrade jx-admin

**Comment by SrutiMi** [comment]
I want to upgrade jx-pipeline.

**Comment by SrutiMi** [comment]
I want to upgrade jx-updatebot

**Comment by ADITYADAS1999** [comment]
Hey @ankitm123 , this  isuue is still open to work?

**Comment by aahmed7** [comment]
@ADITYADAS1999 check the list in the first post. the repos that are not marked need to be updated. Once a repo gets updated, a bot automatically updates the list.

**Comment by ADITYADAS1999** [comment]
> @ADITYADAS1999 check the list in the first post. the repos that are not marked need to be updated. Once a repo gets updated, a bot automatically updates the list.

Thanks for the information @aahmed7

**Comment by rutviklhase** [comment]
I want to upgrade jx-test

**Comment by sudiptob2** [comment]
I want to upgrade jx-promote

**Comment by sudiptob2** [comment]
I want to upgrade jx-pipeline

**Comment by sudiptob2** [comment]
I want to upgrade jx-updatebot

**Comment by sudiptob2** [comment]
jx-promote has been upgraded to 1.19.5

**Comment by sudiptob2** [comment]
I want to upgrade jx-scm

**Comment by Naik-Bharat** [comment]
I want to upgrade jx-slack

**Comment by Vishal1297** [comment]
I want to upgrade [jx-git-operator](https://github.com/jenkins-x/jx-git-operator)

**Comment by Vishal1297** [comment]
I want to upgrade [jx-application](https://github.com/jenkins-x-plugins/jx-application)

**Comment by msvticket** [comment]
This PR has been superseded by #8670. If you want to help out with upgrading to go 1.22, please head over to that PR.
